closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1706,6 +1725,32 @@ private static void checkProviderFormat(String providerInfo, boolean update) {
                 printStackTraceAndExit("Provider format is incorrect: " + providerInfo);
             }
         }
+
+        private void loadProfileCheck(String profileID) {
+            Enumeration<?> pNames = securityProps.propertyNames();
+            while (pNames.hasMoreElements()) {
+                String name = (String) pNames.nextElement();
+                if (name.startsWith(profileID + '.') && name.contains(".jce.provider.")) {
+                    profileCheckProviderNames.add(name);
+                } else if (name.startsWith(profileID + '.')) {
+                    profileCheckPropertyNames.add(name);
+                }
+            }
+        }
+
+        private void checkProfileCheck(String profileID) {
+            if (!profileCheckProviderNames.isEmpty()) {
+                printStackTraceAndExit(
+                        "The order of providers in profile " + profileID
+                                + " (or its base profile) is incorrect.");
+            }
+            if (!profileCheckPropertyNames.isEmpty()) {
+                printStackTraceAndExit(
+                        "The property name: " + profileCheckPropertyNames.toString()
```

Comment:
This message should be reworded to reflect the fact that one or more property names are mentioned.

I'm not sure I agree that it signals misspelling: do we really want to complain and _exit_ if extra, unused properties are specified?

Reply:
For which property names can be recognized, they are actually all hard coded in the `loadProperties()` method. So, if there are property name in the profile which can not be recognized during the initialization, that means, the property name either be misspelled, or the property name is not a restricted security profile property at all and shouldn’t be in the profile. So, we exit and print out the error, and then the client can update their java.security file to correct it.